### DISCUSSION TITLE
How to set interval?

### DISCUSSION THREAD

**Discussion by New-three** [arendst/Tasmota#16058] [Q&A] [answered]
Hello,

I connected a Sonoff TH10 with a temperature sensor to a pump that switches a water circuit for irrigation.
Now I would like to regulate the pump time as in the following scenarios. I found a lot about this on this internet.
But I don't quite understand how to implement it. In the time settings I find the item Rules. I understood it so far that rules can be created.
However, I don't know how complex this can look like.
I have the following ideas/wishes
I would be happy if someone had examples.

- Pump turns on every 40 minutes for 20 minutes. (So ​​within an hour X minutes)
-The pump switches on permanently at a water temperature of over 28 degrees, otherwise as above within an hour X minutes.
-The pump switches on when the water temperature is over 28 degrees for X minutes until the water is below 24 degrees

**Comment by barbudor** [comment, marked as answer]
Documentation for Rules are here: https://tasmota.github.io/docs/Rules/

You probably will need to use triggers such as 'on time#minute|60' for every hour, 'on yoursensor#temperature>28' and 'on yousensor#temperature<24' for managing temperature and variables as a state machine
